File: BOJ/string/boj7785.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;

public class boj7785 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());

        HashMap<String, String> map = new HashMap<>();
        ArrayList<String> arr = new ArrayList<>();

        while (N-- > 0) {
            String[] member = br.readLine().split(" ");
            map.put(member[0], member[1]);
        }

        for (String key : map.keySet()) {
            if (map.get(key).equals("enter")) {
                arr.add(key);
            }
        }

        Collections.sort(arr);

        for (int i = arr.size()-1; i > -1; i--) {
            System.out.println(arr.get(i));
        }
    }
}

File: new-algorithm/src/javaalgo/haulzzang/week15/boj7785.kt
package javaalgo.haulzzang.week15

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.StringTokenizer

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val n: Int = Integer.parseInt(readLine())

    val map: MutableMap<String, String> = mutableMapOf()
    for (i in 1..n) {
        val st = StringTokenizer(readLine())
        val name = st.nextToken()
        val type = st.nextToken()

        map[name] = type
    }

    val answerList: MutableList<String> = mutableListOf()
    for (entry in map) {
        if (entry.value.equals("enter")) {
            answerList.add(entry.key)
        }
    }

    for (ans in answerList.sortedDescending()) {
        println(ans)
    }
}